### ISSUE TITLE
Themes background don't stick with local images

### ISSUE THREAD

**Issue by enoch85** [owncloud/core#23239]
### Steps to reproduce
1. Upgrade from 8.2.2 to 9.0

### Expected behaviour
The themes background image should stay the same

### Actual behaviour
The themes background image are not used in the theme (visible) if it's local, but works with remote address.

http://pastebin.com/mNQDcfcP

### Server configuration
**Operating system**:
Ubuntu Server 14.04
**Web server:**
Apache 2.4
**Database:**
MySQL 5.5
**PHP version:**
5.5
**ownCloud version:** (see ownCloud admin page)
9.0
**Updated from an older ownCloud or fresh install:**
Updated from 8.2.2
**Where did you install ownCloud from:**
ownCloud repos
**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**List of activated apps:**

```
Enabled:
  - activity: 2.2.1
  - calendar: 1.0
  - comments: 0.2
  - contacts: 1.0.0.0
  - dav: 0.1.5
  - documents: 0.12.0
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - mail: 0.3.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - updatenotification: 0.1.0
Disabled:
  - encryption
  - external
  - files_external
  - templateeditor
  - user_external
  - user_ldap
```

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "ocdd06e12069",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.fsgo.se",
            "cloud.fsgo.eu",
            "fsgo.se",
            "www.fsgo.se"
        ],
        "datadirectory": "\/var\/www\/owncloud\/data",
        "overwrite.cli.url": "https:\/\/cloud.fsgo.se\/owncloud",
        "dbtype": "mysql",
        "version": "9.0.0.19",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
        "default_language": "sv",
        "allow_user_to_change_display_name": true,
        "mail_smtpmode": "smtp",
        "mail_smtpauth": 1,
        "mail_smtphost": "mail.fsgo.se",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "admin",
        "mail_domain": "fsgo.se",
        "forcessl": true,
        "mail_smtpsecure": "tls",
        "mail_smtpauthtype": "LOGIN",
        "theme": "fsgo",
        "maintenance": false,
        "secret": "***REMOVED SENSITIVE VALUE***",
        "forceSSLforSubdomains": true,
        "loglevel": 2,
        "app.mail.server-side-cache.enabled": true,
        "app.mail.imaplog.enabled": true,
        "logtimezone": "Europe\/Stockholm",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": "true",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "\/var\/run\/redis.sock",
            "port": 0,
            "timeout": 0,
            "dbindex": 0
        },
        "appstore.experimental.enabled": true,
        "trashbin_retention_obligation": "30",
        "updatechecker": false
    }
}

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
Chrome 49
**Operating system:**
Deepin 15 and Windows (tested both)
#### Web server error log
```
No error log
```

#### ownCloud log (data/owncloud.log)
```
No errors
```

#### Browser log
```
No error log
```

![deepinscreenshot20160314175749](https://cloud.githubusercontent.com/assets/4511254/13752293/58349b84-ea0e-11e5-8f0e-94ad9e40bad7.png)

**Comment by PVince81** [comment]
You mean relative vs absolute URLs ?

Can you click on that "../img/back.jpg" and find out what the full URL is ?
Usually the relative URL is relative to the CSS file itself. So if "styles.css" is inside the themes folder, then the path is relative to that.

**Comment by 20xx** [comment]
Hi, I'm having exactly the same issue described.
The background image reference wasn't changed when updating from 8.2.2 to 9.0.

Is this related to Content Security Policy?

**Comment by enoch85** [comment]
o@PVince81 I use the exact same settings as in 8.2.2. 

``` 
drwxr-x--- 2 root www-data  4096 Oct 23 22:37 ./
drwxr-x--- 4 root www-data  4096 Dec  6 00:50 ../
-rw-r----- 1 root www-data 12519 Oct  4 16:22 apps.css
-rw-r----- 1 root www-data   913 Dec  6 00:58 header.css
-rw-r----- 1 root www-data   745 Mar 10 02:15 styles.css
root@en0ch:/var/www/owncloud/themes/custom/core/css# ll /var/www/owncloud/themes/custom/core/img/
total 504
drwxr-x--- 3 root www-data   4096 Mar 10 02:10 ./
drwxr-x--- 4 root www-data   4096 Dec  6 00:50 ../
drwxr-x--- 2 root www-data   4096 Jan  6 23:57 actions/
-rw-r----- 1 root www-data 363682 Sep 21  2015 back.jpg
-rw-r----- 1 root www-data   2910 Sep 21  2015 logo-icon.svg
-rw-r----- 1 root www-data   5051 Oct 21 03:50 logo.svg
-rw-r----- 1 root www-data 123125 Mar 10 02:10 tech.jpg
root@en0ch:/var/www/owncloud/themes/custom/core/css# 
``` 

Have you changed something in 9.0?

**Comment by PVince81** [comment]
I'm not aware of any theming related changes, but often there are security hardenings so it could be that.

@enoch85 @20xx can you guys check the network console and find the call that retrieves your background image from the theme ? Goal is to find out what the return code is and what the full URL is.
Either the URL is wrong, or it's blocked for some reason (wrong permissions, .htaccess restrictions, etc)

**Comment by 20xx** [comment]
![bildschirmfoto 2016-03-23 um 14 20 14](https://cloud.githubusercontent.com/assets/222474/13986496/0116a59e-f103-11e5-8322-be3f32d984d0.png)
![bildschirmfoto 2016-03-23 um 14 22 07](https://cloud.githubusercontent.com/assets/222474/13986497/0118eafc-f103-11e5-8b31-c6ae1c8a02a4.png)

The Full URL is correct, but I cannot load the image via this full URL directly (being redirected to login page)

In Firebug I receive this error on the landing page:

Content Security Policy: Settings of this page have blocked loading a resource on self
 ("script-src https://<owncloud-instance> 'unsafe-eval'").
onsubmit attribute on DIV element

Thanks for your support, Matthias

**Comment by PVince81** [comment]
If you're getting a redirect when trying to access the full URL of the image, then either the permissions are wrong in your themes folder (the web server needs to be able to access it) or your web server config or .htaccess config is forbidding it somehow.

**Comment by 20xx** [comment]
![bildschirmfoto 2016-03-23 um 14 31 03](https://cloud.githubusercontent.com/assets/222474/13986672/fb382ff2-f103-11e5-9e39-eb00b02a3055.png)
![bildschirmfoto 2016-03-23 um 14 31 16](https://cloud.githubusercontent.com/assets/222474/13986673/fb3f0548-f103-11e5-9a5c-4e3aa16a60ce.png)
![bildschirmfoto 2016-03-23 um 14 31 26](https://cloud.githubusercontent.com/assets/222474/13986674/fb40e4bc-f103-11e5-8960-3cc62c230a83.png)

Thanks, but I can open styles.css directly with the same permissions as sbit_one.jpg ...

**Comment by enoch85** [comment]
@PVince81 As I said earlier, nothing changed except the upgrade to 9.0 stable. I see this on several instances, not just mine.

The permissions are theese: https://github.com/enoch85/ownCloud-VM/blob/master/production/setup_secure_permissions_owncloud.sh

**Comment by 20xx** [comment]
![bildschirmfoto 2016-03-26 um 09 20 40](https://cloud.githubusercontent.com/assets/222474/14059040/2d83a428-f334-11e5-9457-e65cf15da90e.png)

Hi there, it works now. I had to adapt the main .htaccess file in the root directory.
See screenshot:

# Rewrite rules for `front_controller_active`

Is it possible to generally add "jpg" to
RewriteCond %{REQUEST_FILENAME} !\.(css|jpg|js|svg|gif|png|html|ttf|woff|ico)$

Best, Matthias

**Comment by enoch85** [comment]
@PVince81 Yes, this should go in the default settings of .htaccess as it breaks all the themes with custom background images otherwise.

I can confirm that the fix works. Thanks @20xx!

**Comment by MorrisJobke** [comment]
Fix for 9.1 is in https://github.com/owncloud/core/pull/23129

Backport for stable9.0: #23590
